src/_pytest/fixtures.py

```diff
@@ -641,11 +642,8 @@ def _compute_fixture_value(self, fixturedef: "FixtureDef[object]") -> None:
 
         # Check if a higher-level scoped fixture accesses a lower level one.
         subrequest._check_scope(argname, self._scope, scope)
-        try:
-            # Call the fixture function.
```

Comment:
This looks like it originally ensured teardown even when setup failed

I suspect that we are missing a edge case tests there

Reply:
The specific schedule is slightly better,

Reply:
On further consideration, changing that seems... bad to me? It means we'll be running the finalizer multiple times for a fixture with a cached exception, even if its setup was only run once. That *is* how it worked in the past though

Reply:
Apologies, what I meant is that a single schedule is better than the current mechanism

Ideally this logic would move to a data structure instead of the code where it is

Reply:
Right - no yeah I agree that the current implementation is quite messy and would benefit from an overhaul, but that seems out of scope for this PR.
I'm not sure I follow your original comment in that case, the `try/except` that previously was on this line is now moved one step deeper into `execute()` - and so there should be no change wrt to behaviour of scheduling finalizer if setup fails.

The only real change should :tm: be that the finalizer isn't scheduled if the value is cached (regardless of if it's an exception or not). If the setup code for the fixture fails, it's catched by the new `try/finally`.
If some other unrelated code in `execute` raises an exception the finalizer will not be scheduled, but in that case the setup has not run either.

Reply:
So is there a problem I should address, or was your comment just a musing on how this should be generally overhauled? Or do you consider that a requirement before modifying any logic?